src/controllers/user.ts

```diff
@@ -227,6 +227,35 @@ const userController = {
         }
     },
 
+    resetLastSubmission: async (req: Request, res: Response) => {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            res.status(statusCodes.MISSING_PARAMS).json(
+                errors.formatWith(errorMessage).array()[0]
+            );
+        } else {
+            try {
+                const { userId } = req.params;
+                await userDBInteractions.resetLastSubmission(userId);
+                const user: IUserModel = await userDBInteractions.find(userId);
+                if (!user)
+                    res.status(statusCodes.NOT_FOUND).json({
+                        status: statusCodes.NOT_FOUND,
+                        message: "User not found"
+                    });
+                else {
+                    const updatedUser: IUserModel = await userDBInteractions.update(
+                        userId,
+                        user
+                    );
```

Comment:
Why are you updating here if the "user" variable is unchanged? I suggest you either call the new resetLastSubmission DB interaction here or you manually update the "user" by setting the lastSubmission to {} then calling the update DB interaction (I prefer the second option since it allows us to reuse an premade DB interaction instead of making a new one).